NetworkPkg/NetworkPcds.dsc.inc

```diff
@@ -11,6 +11,8 @@
 #
 ##
 
+!if $(NETWORK_ENABLE) == TRUE

```

Comment:
Not sure if this is the right place to put it. Ideally you would check if network was enabled before including the NetworkPkg.dsc which includes this file, right?

Reply:
Yeah, your suggestion is the ideal case. However, note that many of the Network includes are currently being added without any checks. In this context, then we also need to make sure that the network allows HTTP secure connections (`NETWORK_ALLOW_HTTP_CONNECTIONS`) NOT within this file.

If we take the approach of checking before includes, we would have to modify more than five DSC files:

```
[triste@reeva edk2]$ grep -rni -A 2 -B 2 "\!include NetworkPkg/NetworkPcds.dsc.inc" *
grep: warning: stray \ before !
ArmVirtPkg/ArmVirtQemuKernel.dsc-160-  # Network Pcds
ArmVirtPkg/ArmVirtQemuKernel.dsc-161-  #
ArmVirtPkg/ArmVirtQemuKernel.dsc:162:!include NetworkPkg/NetworkPcds.dsc.inc
ArmVirtPkg/ArmVirtQemuKernel.dsc-163-
ArmVirtPkg/ArmVirtQemuKernel.dsc-164-  gEfiMdeModulePkgTokenSpaceGuid.PcdResetOnMemoryTypeInformationChange|FALSE
--
ArmVirtPkg/ArmVirtQemu.dsc-195-  # Network Pcds
ArmVirtPkg/ArmVirtQemu.dsc-196-  #
ArmVirtPkg/ArmVirtQemu.dsc:197:!include NetworkPkg/NetworkPcds.dsc.inc
ArmVirtPkg/ArmVirtQemu.dsc-198-
ArmVirtPkg/ArmVirtQemu.dsc-199-  # System Memory Base -- fixed at 0x4000_0000
--
NetworkPkg/NetworkPcds.dsc.inc-3-#
NetworkPkg/NetworkPcds.dsc.inc-4-# This file can be included to the [Pcds*] section(s) of a platform DSC file
NetworkPkg/NetworkPcds.dsc.inc:5:# by using "!include NetworkPkg/NetworkPcds.dsc.inc" to specify PCD settings
NetworkPkg/NetworkPcds.dsc.inc-6-# according to the value of flags described in "NetworkDefines.dsc.inc".
NetworkPkg/NetworkPcds.dsc.inc-7-#
--
NetworkPkg/Network.dsc.inc-27-
NetworkPkg/Network.dsc.inc-28-[PcdsFixedAtBuild]
NetworkPkg/Network.dsc.inc:29:!include NetworkPkg/NetworkPcds.dsc.inc
NetworkPkg/Network.dsc.inc-30-
NetworkPkg/Network.dsc.inc-31-[LibraryClasses]
--
OvmfPkg/Bhyve/BhyveX64.dsc-507-  # Network Pcds
OvmfPkg/Bhyve/BhyveX64.dsc-508-  #
OvmfPkg/Bhyve/BhyveX64.dsc:509:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/Bhyve/BhyveX64.dsc-510-
OvmfPkg/Bhyve/BhyveX64.dsc-511-  # Point to the MdeModulePkg/Application/UiApp/UiApp.inf
--
OvmfPkg/OvmfPkgIa32X64.dsc-590-  # Network Pcds
OvmfPkg/OvmfPkgIa32X64.dsc-591-  #
OvmfPkg/OvmfPkgIa32X64.dsc:592:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/OvmfPkgIa32X64.dsc-593-
OvmfPkg/OvmfPkgIa32X64.dsc-594-  gEfiShellPkgTokenSpaceGuid.PcdShellFileOperationSize|0x20000
--
OvmfPkg/CloudHv/CloudHvX64.dsc-558-  # Network Pcds
OvmfPkg/CloudHv/CloudHvX64.dsc-559-  #
OvmfPkg/CloudHv/CloudHvX64.dsc:560:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/CloudHv/CloudHvX64.dsc-561-
OvmfPkg/CloudHv/CloudHvX64.dsc-562-  gEfiShellPkgTokenSpaceGuid.PcdShellFileOperationSize|0x20000
--
OvmfPkg/OvmfPkgIa32.dsc-571-  # Network Pcds
OvmfPkg/OvmfPkgIa32.dsc-572-  #
OvmfPkg/OvmfPkgIa32.dsc:573:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/OvmfPkgIa32.dsc-574-
OvmfPkg/OvmfPkgIa32.dsc-575-  gEfiShellPkgTokenSpaceGuid.PcdShellFileOperationSize|0x20000
--
OvmfPkg/OvmfPkgX64.dsc-603-  # Network Pcds
OvmfPkg/OvmfPkgX64.dsc-604-  #
OvmfPkg/OvmfPkgX64.dsc:605:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/OvmfPkgX64.dsc-606-
OvmfPkg/OvmfPkgX64.dsc-607-  gEfiShellPkgTokenSpaceGuid.PcdShellFileOperationSize|0x20000
--
OvmfPkg/Microvm/MicrovmX64.dsc-561-  # Network Pcds
OvmfPkg/Microvm/MicrovmX64.dsc-562-  #
OvmfPkg/Microvm/MicrovmX64.dsc:563:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/Microvm/MicrovmX64.dsc-564-
OvmfPkg/Microvm/MicrovmX64.dsc-565-  gEfiShellPkgTokenSpaceGuid.PcdShellFileOperationSize|0x20000
--
OvmfPkg/LoongArchVirt/LoongArchVirtQemu.dsc-374-  # Network Pcds
OvmfPkg/LoongArchVirt/LoongArchVirtQemu.dsc-375-  #
OvmfPkg/LoongArchVirt/LoongArchVirtQemu.dsc:376:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/LoongArchVirt/LoongArchVirtQemu.dsc-377-
OvmfPkg/LoongArchVirt/LoongArchVirtQemu.dsc-378-  gEfiMdeModulePkgTokenSpaceGuid.PcdFlashNvStorageVariableSize         | 0x40000
--
OvmfPkg/OvmfXen.dsc-442-  # Network Pcds
OvmfPkg/OvmfXen.dsc-443-  #
OvmfPkg/OvmfXen.dsc:444:!include NetworkPkg/NetworkPcds.dsc.inc
OvmfPkg/OvmfXen.dsc-445-
OvmfPkg/OvmfXen.dsc-446-!ifdef $(DEBUG_ON_HYPERVISOR_CONSOLE)
```

Additionally, there are already include files that contain statements for checks, such as [Network.fdf.inc](https://github.com/tianocore/edk2/blob/master/NetworkPkg/Network.fdf.inc), [NetworkBuildOptions.dsc.inc](https://github.com/tianocore/edk2/blob/master/NetworkPkg/NetworkBuildOptions.dsc.inc), [NetworkComponents.dsc.inc](https://github.com/tianocore/edk2/blob/master/NetworkPkg/NetworkComponents.dsc.inc), and [NetworkDefines.dsc.inc](https://github.com/tianocore/edk2/blob/master/NetworkPkg/NetworkDefines.dsc.inc).

Reply:
I think thats fair. I see similar conditions for NETWORK_ENABLE used in the files you mentioned above.